### PR TITLE
added Buffer.MemoryCopy to ArrayCopy benchmarks

### PR DESCRIPTION
**Pull request by adamsitnik**
I added `Buffer.MemoryCopy` to benchmarks and also fixed the way `Unsafe` was used. 

```ini

Host Process Environment Information:
BenchmarkDotNet-Dev.Core=v0.9.9.0
OS=Microsoft Windows NT 6.2.9200.0
Processor=Intel(R) Core(TM) i7-4700MQ CPU 2.40GHz, ProcessorCount=8
Frequency=2338336 ticks, Resolution=427.6545 ns, Timer=TSC
CLR=MS.NET 4.0.30319.42000, Arch=64-bit RELEASE [RyuJIT]
GC=Concurrent Workstation
JitModules=clrjit-v4.6.1055.0

Type=ArrayCopy  Mode=Throughput  

```
                          Method | Count |      Median |     StdDev | Scaled | Scaled-SD | Gen 0 | Gen 1 | Gen 2 | Bytes Allocated/Op |
-------------------------------- |------ |------------ |----------- |------- |---------- |------ |------ |------ |------------------- |
         **Copy using Array.Copy()** |   **100** |  **30.5768 ns** |  **0.1895 ns** |   **1.00** |      **0.00** |     **-** |     **-** |     **-** |               **0,00** |
            Copy using Unsafe<T> |   100 |  10.0101 ns |  0.3138 ns |   0.33 |      0.01 |     - |     - |     - |               0,00 |
 Copy using Buffer.MemoryCopy<T> |   100 |  10.4664 ns |  0.4898 ns |   0.35 |      0.02 |     - |     - |     - |               0,00 |
         **Copy using Array.Copy()** |  **1000** |  **41.5319 ns** |  **4.1919 ns** |   **1.00** |      **0.00** |     **-** |     **-** |     **-** |               **0,00** |
            Copy using Unsafe<T> |  1000 |  43.0451 ns |  2.0176 ns |   1.03 |      0.10 |     - |     - |     - |               0,00 |
 Copy using Buffer.MemoryCopy<T> |  1000 |  41.5373 ns |  6.3491 ns |   0.98 |      0.17 |     - |     - |     - |               0,00 |
         **Copy using Array.Copy()** | **10000** | **215.7254 ns** | **28.6357 ns** |   **1.00** |      **0.00** |     **-** |     **-** |     **-** |               **0,06** |
            Copy using Unsafe<T> | 10000 | 405.6742 ns |  9.6347 ns |   2.06 |      0.35 |     - |     - |     - |               0,11 |
 Copy using Buffer.MemoryCopy<T> | 10000 | 210.8156 ns | 30.6141 ns |   0.98 |      0.23 |     - |     - |     - |               0,06 |

### CODE REVIEW

File: tests/ImageProcessorCore.Benchmarks/General/ArrayCopy.cs [hunk 1]
```diff
@@ -7,25 +7,44 @@ namespace ImageProcessorCore.Benchmarks.General
...
+            Array.Copy(source, destination, Count);
         }
 
         [Benchmark(Description = "Copy using Unsafe<T>")]
-        public unsafe double CopyUnsafe()
+        public unsafe void CopyUnsafe()
         {
-            double[] destination = new double[10000];
-            Unsafe.Copy(Unsafe.AsPointer(ref destination), ref source);
+            fixed (byte* pinnedDestination = destination)
```
Comment (JimBobSquarePants) [inline tests/ImageProcessorCore.Benchmarks/General/ArrayCopy.cs]: I wonder whether fixing makes a difference here? In the CoreFX unit tests they are using `stackalloc` instead up to 100,000. https://github.com/dotnet/corefx/blob/98b6351acaa96effd3f34a1e3fb7cef9127fa72d/src/System.Runtime.CompilerServices.Unsafe/tests/UnsafeTests.cs#L194

File: tests/ImageProcessorCore.Benchmarks/General/ArrayCopy.cs [hunk 2]
```diff
@@ -7,25 +7,44 @@ namespace ImageProcessorCore.Benchmarks.General
...
-        public double CopyArray()
-        {
+        byte[] source, destination;
 
-            double[] destination = new double[10000];
-            Array.Copy(source, destination, 10000);
+        long sizeInBytes;
 
-            return destination[0];
+        [Setup]
```
Comment (JimBobSquarePants) [inline tests/ImageProcessorCore.Benchmarks/General/ArrayCopy.cs]: Is there a teardown equivalent to this? I've been trying to figure out how to do cleanup for a while.

Comment (adamsitnik) [inline tests/ImageProcessorCore.Benchmarks/General/ArrayCopy.cs]: @JimBobSquarePants It was [added recently](https://github.com/PerfDotNet/BenchmarkDotNet/issues/215), it's called `Cleanup`.

File: tests/ImageProcessorCore.Benchmarks/General/ArrayCopy.cs (hunk 1 above)
Comment (adamsitnik) [inline tests/ImageProcessorCore.Benchmarks/General/ArrayCopy.cs]: @JimBobSquarePants As far as I know `Unsafe.AsPointer(ref destination)` gives you pointer to the array (object), `fixed (byte* pinnedSource = source)` gives you pointer to the first object in the array. This is tricky ;)

Comment (JimBobSquarePants) [inline tests/ImageProcessorCore.Benchmarks/General/ArrayCopy.cs]: Interesting... I wasn't aware of that.

File: tests/ImageProcessorCore.Benchmarks/General/ArrayCopy.cs (hunk 2 above)
Comment (JimBobSquarePants) [inline tests/ImageProcessorCore.Benchmarks/General/ArrayCopy.cs]: Yussssss!

**Pull request merged**